components/script/dom/servoparser/mod.rs

```diff
@@ -214,6 +214,7 @@ impl ServoParser {
             loader,
             None,
             None,
+            context_document.status_code(),
```

Comment:
Since this is a synthetic document, why do we want to propagate another document's status code?